cpp/src/arrow/ipc/reader.h

```diff
@@ -301,9 +323,39 @@ class ARROW_EXPORT CollectListener : public Listener {
     return record_batches_;
   }
 
+  /// \return the all decoded metadatas
+  std::vector<std::shared_ptr<KeyValueMetadata>> metadatas() const { return metadatas_; }
```

Comment:
We may want to use better name for a vector of metadata...

Reply:
I think this is ok.